sig-and-wg/sig-core/proposals/generating-changelog-with-pr-labels.md

```diff
@@ -0,0 +1,105 @@
+- Pull requests should have a title written in imperative mood (like commit messages). The title will be included in changelog, along with the author and a link to the particular PR.
+- No label on a pull request means that the change won't be included in changelog.
+
+### Result
+
+- The changelog is written to CHANGELOG.md file and put in GitHub release description via [GitHub API](https://developer.github.com/v3/repos/releases/#create-a-release) for a specific git tags
+- All changes are grouped by selected labels
+
+Desired output would look like this:
+    - [example GitHub releases](https://github.com/lerna/lerna-changelog/releases)
+    - [example CHANGELOG.md](https://github.com/lerna/lerna-changelog/blob/master/CHANGELOG.md)
+
+### Labels
+
+Every label listed below is taken into account during changelog generation process.
+
+For now we don't introduce any new labels. We reuse `area/` labels from [the accepted proposal](https://github.wdf.sap.corp/SAP-CP-Extension-Factory/community/blob/master/sig-and-wg/wg-github-issues-migration/proposals/github-issues-labels-proposal.md). This means that all changes are grouped by area (e.g. `installation`, `security`...).
+
+**Example:**
+
+> ### Installation
+> - Sample change 1
+> - Sample change 2
+>
+> ### Security
+> - Sample change 1
+> - Sample change 2
+
+In future we can introduce nested grouping for changelog (first level: change type, second one: area of the change). Change type will marked with additional prefixed labels. Choosing right prefix will be discussed later.
+
+**Example:**
+
+> ### Bug fixes
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+>
+> ### Features
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+
+## Pros
+
+- One place to define general changes in pull request: pull request title, which is usually prefilled with commit message, if there is just only one commit
+- Selecting type of change is a no-brainer: contributor expands the list of available PR labels and chooses right category.
+- We can filter unlabeled pull requests ([link](https://github.com/kyma-project/kyma/issues?q=is%3Aopen+is%3Apr+no%3Alabel)) and easily add labels to them to generate complete changelog. Also, it's easy to see the full list of PRs to check if they are named correctly (and adjust titles - we can do that even for closed pull requests)
+- It enforces user to make a single type of change in a pull request
+- There are many tools available for generating changelog from pull requests labels.
+
+## Cons
+
+- More detailed changes in changelog. For example, when there is a bigger feature, which needs multiple PRs, it will be visible as multiple bullet points.
+- Pull request authors and reviewers can forgot about adding a label to categorize the change. Anyway, even closed pull requests can be edited before generating the changelog. 
```

Comment:
We could consider a rule to reject merging PRs that don't have labels relevant for changelog (not necessarily now, but it could already be mentioned here as a possible future enhancement?)

Reply:
That would be my suggestion. Not something that needs to be done now, but kept somewhere so we don't forget. Thanks!

Reply:
Good idea. Done!